server/src/main/java/org/elasticsearch/index/mapper/ContentPath.java

```diff
@@ -24,6 +24,11 @@ public ContentPath() {
         this.sb = new StringBuilder();
     }
 
+    protected String[] getPath() {
+        // used for testing
```

Comment:
I added this method because I think making the code testable is better than using hacks (Reflection).
But I am open to discuss on this!

Reply:
It doesn't have to be `protected` but 'pkg-private' is acceptable.